src/DependencyRewriterPlugin.php

```diff
@@ -115,8 +118,14 @@ public function onPreCommandRun(PreCommandRunEvent $event)
     public function onPreDependenciesSolving(InstallerEvent $event)
     {
         $this->output(sprintf('<info>In %s</info>', __METHOD__), IOInterface::DEBUG);
-        $request = $event->getRequest();
-        $jobs = $request->getJobs();
+        if (method_exists($event, 'getRequest')) {
+            $request = $event->getRequest();
+            $jobs = $request->getJobs();
+        } else {
+            $request = new Request(new LockArrayRepository());
+            $jobs = $request->getRequires();
```

Comment:
I'll check this out but just by seeing this, I cant think that this will return the same as in v1 `$requests->getJobs()` 🤔

Reply:
Okay, I've dig a bit more into the magic of this library.

We have to totally rewrite it for composer 2.0 as its not about "detection" but to changing composer internals so it does continue with the replaced packages.

Reply:
thank you ;)